Makefile

```diff
@@ -285,18 +288,18 @@ ifneq "$(origin BUILDER_IMAGE_DOCKERFILE)" "file"
 	@echo "Dockerfile for builder image has been overridden"
 	@echo "Skipping push of custom image"
 else
-	docker push $(BUILDER_IMAGE)
+	${BUILDER} push $(BUILDER_IMAGE)
 endif
 
 build-image-hugo:
-	cd site && docker build --pull -t $(HUGO_IMAGE) .
+	cd site && ${BUILDER} build --pull -t $(HUGO_IMAGE) .
 
 clean:
 # if we have a cached image then use it to run go clean --modcache
 # this test checks if we there is an image id in the BUILDER_IMAGE_CACHED variable.
 ifneq ($(strip $(BUILDER_IMAGE_CACHED)),)
 	$(MAKE) shell CMD="-c 'go clean --modcache'"
-	docker rmi -f $(BUILDER_IMAGE) || true
+	${BUILDER} rmi -f $(BUILDER_IMAGE) || true
```

Comment:
`buildx` has `rmi`? I think its only `rm`

Reply:
That is correct! sorry! I will create an IMAGE_TOOL var to decide if just podman or docker to fix this

Reply:
better to set BUILDER just as `docker`, instead of `docker buildx` (if so, better to change variable name?), to change less things?

Reply:
have to use docker buildx when building though :(

Reply:
I think I am a little lost. You mean whenever `bocker build` is called, it should be `docker buildx build`? If yes, then this seems like the only place missing https://github.com/vmware-tanzu/velero/blob/main/Makefile#L292

also makefile has a buildx logic https://github.com/vmware-tanzu/velero/blob/main/Makefile#L77-L81

Reply:
@mateusoliveira43 I think I am confused as well.

The change I am making, so that we can use podman as well as docker. docker has a sub-command `buildx` that allows for creating of manifest list images. Podman is able to to do this without that sub command. 

When building and using docker, one should use `docker buildx build` and when using podman one should use `podman build`. 

When doing other image related tasks you just use `docker` or `podman`. 

Does that help or make the convo more confusing?

Reply:
Alright, got it, thank you.

So, I believe we can simplify this as `buildx` logic is already handled by makefile

we add the new logic to check docker or podman and inside it we need to run the `buildx` logic https://github.com/vmware-tanzu/velero/blob/main/Makefile#L77-L81 (for podman is always false) then just change the `docker` invocations to `podman` (we would also need to change this logic https://github.com/vmware-tanzu/velero/blob/main/Makefile#L183 to not fail to podman, but should be easy)

Do you agree?

Reply:
I think that the current logic is pretty simple no and I don't know if I follow this. Can you submit a PR to my branch if you feel strongly it is better so that I can see it?